packages/groq-builder/README.md

```diff
@@ -0,0 +1,92 @@
+# `groq-builder`
+
+A **schema-aware**, strongly-typed GROQ query builder.  
+It enables you to use **auto-completion** and **type-checking** for your GROQ queries. 
+
+## Features
+
+- **Schema-aware** - Use your `sanity.config.ts` for auto-completion and type-checking
+- **Strongly-typed** - Query results are strongly typed, based on the schema
+- **Optional runtime validation** - Validate or transform query results at run-time, with broad or granular levels
+
+## Example
+
+```ts
+import { createGroqBuilder } from 'groq-builder';
+import type { MySchemaConfig } from './my-schema-config';
+//            ☝️ Note:
+// Please see the "Schema Configuration" docs 
+// for an overview of this SchemaConfig type 
+
+const q = createGroqBuilder<MySchemaConfig>()
+
+const productsQuery = (
+  q.star
+   .filterByType('products')
+   .order('price desc')
+   .slice(0, 10)
+   .projection(q => ({
+     name: true,
+     price: true,
+     slug: q.projection('slug').projection('current'),
+     imageUrls: q.projection('images[]').deref().projection('url')
+   }))
+);
+```
```

Comment:
This is a great point, and I actually have a lot of potential solutions...
1. Allow strongly-typed dot-notation strings, so we can do `slug: q.projection('slug.current'),`
2. Allow strongly-typed dot-notation strings without wrapping, like: `slug: 'slug.current',`
3. In GroqD, I really like how `q` can be called like `q('slug.current')`, because it's so short!  However, it lacks semantics - you can put any groq text into that string.  
So I've been thinking, maybe `q.projection(...)` could be aliased as `q.p(...)` since it's so commonly used?  It's the only method that I think we'd ever alias.  Having said that ... if we implemented 1. and/or 2., then the problem is really minimized, so maybe it's better to be more verbose?  

WDYT?

Reply:
I implemented option 1